js/graviex.js

```diff
@@ -0,0 +1,231 @@
+                '1h': '60',
+                '2h': '120',
+                '4h': '240',
+                '6h': '360',
+                '12h': '720',
+                '1d': '1440',
+                '3d': '4320',
+                '1w': '10080',
+            },
+            'urls': {
+                'logo': '',
+                'api': {
+                    'public': 'https://graviex.net//webapi/v3',
+                    'private': 'https://graviex.net//webapi/v3',
+                },
+                'www': 'https://graviex.net',
+                'doc': 'https://graviex.net/documents/api_v3',
+                'fees': 'https://graviex.net/documents/fees-and-discounts',
+            },
+            'api': {
+                // market: Unique market id. It's always in the form of xxxyyy, where xxx is the base currency code, yyy is the quote currency code, e.g. 'btccny'. All available markets can be found at /api/v2/markets.
+                'public': {
+                    'get': [
+                        'markets',
+                        'tickers',
+                        'trades', // Requires market XXXYYY. Optional: limit, timestamp, from, to, order_by
+                    ],
+                },
+                'private': {
+                },
+            },
+            'fees': {
+                'trading': {
+                    'percentage': true,
+                    'maker': this.parseNumber ('0.001'),
+                    'taker': this.parseNumber ('0.2'),
+                },
+                'funding': {
+                    'withdraw': {
+                        'BTC': 0.0004,
+                        'ETH': 0.022,
+                        'DOGE': 2.0,
+                        'NYC': 1.0,
+                        'XMR': 0.02,
+                        'PIVX': 0.2,
+                        'NEM': 0.05,
+                        'SCAVO': 5.0,
+                        'SEDO': 5.0,
+                        'USDT': 3.0,
+                        'GDM': 0.3,
+                        'PIRL': 0.005,
+                        'PK': 0.1,
+                        'ORM': 10,
+                        'NCP': 10,
+                        'ETM': 10,
+                        'USD': 0,
+                        'EUR': 0,
+                        'RUB': 0,
+                        'other': 0.002,
+                    },
+                },
+            },
+            'limits': {
+                'amount': {
+                    'min': 0.001,
+                    'max': undefined,
+                },
+            },
+            'precision': {
+                'amount': 8,
+                'price': 8,
+            },
+        });
+    }
+
+    async fetchMarkets (params = {}) {
+        const response = await this.publicGetMarkets ();
+        const markets = response;
+        const result = [];
+        for (let i = 0; i < markets.length; i++) {
+            const id = markets[i]['id'];
+            const symbolParts = markets[i]['name'].split ('/');
```

Comment:
We should always use safe-methods for all keys:

- https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#sanitizing-input-with-safe-methods

Reply:
OK. Fixed